orchestrator/README.md

```diff
@@ -79,3 +80,22 @@ be omitted if the default behavior is desired.
 The orchestrator runs as a non-root user. As such, the container must be run with permissions of the `docker` group on the host
 in order to access the socket. The socket must also be mounted into the container (the bind mount `/var/run/docker.sock:/var/run/docker.sock:ro`).
 This docker group ID changes from system to system and needs to be checked manually using a command like `cat /etc/group | grep docker | awk -F: '{ print $3 }'`
+
+## Cache Implementation
+
+IMAGE uses Memcached as in-memory data store. Cache is implemented using [MemJS](https://www.npmjs.com/package/memjs). Following is the confugration to enable Cache for preprocessors:
+
+- Cache size is configured in the docker-compose in the commad attribute under memcached service `command: -m 4096` implies cache size of 4GB.
+
+- Cache timeout is configured at the preprocessor level, with the label `ca.mcgill.a11y.image.cacheTimeout` . Label value is the timeout value in seconds. Timeout value of 0 indicates that Cache is disabled for a preprocessor. Missing `ca.mcgill.a11y.image.cacheTimeout` label on the preprocessor will default to timeout value of 0.
+
+- Cache key is generated using the following attributes:
+  - `reqData` can have the following falues
+    - `request["data"]` (for graphics)
+    - `request["placeID"]`/`request["coordinates"]` (for maps)
+    - `request["highChartsData"]` (for charts)
+  - `preprocessor` - preprocessor id as returned in the response
+  - `debugMode` - true/false depending on if the debug mode is enabled or not.
```

Comment:
Looking at this again, is there actually a case where debug mode _should_ impact the results of a preprocessor? I can't think of anything and am leaning towards this being removed. Thoughts, @jeffbl?

Reply:
Good question. The idea when we first discussed was that preprocessors (or eventually handlers, if we end up caching any of those in the future) might change their behavior based on the debug (or other capabilities) flags. But as @JRegimbal pointed out in conversation this morning, if the preprocessors are, say, outputting more log information when they see the debug flag, that won't appear past the first invocation anyway, if we pay attention to the debug flag. If preprocessors are changing their actual results, then it really isn't a good proxy for actual use, and it should really be some other flag other than "debug". I feel like there may be a valid reason to do this, but until we can articulate it, agree debug can be removed.

However, this also begs the question of other capabilities. I could argue it would be easiest to simply include all capabilities (including debug). Worst case, e.g., for debug, we do one extra uncached runthrough, to be sure that if someone does do something based on any capability, they don't get an incorrect cached version. At least this way the story is consistent - any capability difference causes a cache miss. Once we start picking and choosing, even for good reasons, it becomes more difficult for developers to know what to expect, if they do things we don't anticipate. @jaydeepsingh25 thoughts?